### ISSUE TITLE
Consider using a different extension than `.snap`

### ISSUE THREAD

**Issue by spenserblack** [mitsuhiko/insta#425]
Right now insta's snapshot files share an extension with Jest's snapshot files. This is a low priority issue, more of a small annoyance than a problem, but this causes GitHub (via [linguist](https://github.com/github-linguist/linguist)) to detect all of insta's snapshots as Jest snapshots, and apply invalid syntax highlighting. If you ever plan on having a breaking change, perhaps you'd want to transition over to using a new extension, like `.insta`, to make it easier for tools to know that Insta snapshots are *not* Jest snapshots.

On the other side of things, one might consider registering insta snapshots as a language for GitHub to support, and adding a heuristic (perhaps the front matter can be used?) to help differentiate them.

Right now, as a workaround, I add the following to `.gitattributes` to remove any awkward syntax highlighting and inaccurate language statistics.
```gitattributes
*.snap linguist-language=Text
```

**Comment by max-sixty** [comment]
keen for ideas of how to manage this — I don't have an immediate view on this end...

**Comment by Enselic** [comment]
> keen for ideas of how to manage this — I don't have an immediate view on this end...

I think changing extension to something not already taken is the best long term solution. Like `.insta` that was suggested above.

**Comment by spenserblack** [comment]
Besides using a different extension, *if this file format is popular enough,* you can make a PR into Linguist to detect Insta snapshots as a language, and provide a heuristic to tell the two formats apart.

For allowing the fix to *automatically* apply to GH repos, I think there are only two options (not mutually exclusive):
1. Pick a completely unique file extension that other popular tools don't use
2. Add Insta Snapshots as a known language to Linguist

You'd need some proof that insta is fairly popular if you plan on making a PR to Linguist. Since a search for `path:*.snap` obviously won't work, I think the best search for proof of popularity would be `path:Cargo.toml "insta ="`

**Comment by Enselic** [comment]
That'd only be a fix for GitHub though. To solve it "once and for all" we'd need to change extension. Doing that kind of fix for all source code hosting services and IDEs does not scale and is never finished  since new services and IDEs keep popping up.

**Comment by spenserblack** [comment]
Just want to be clear that my vote is for changing the extension -- just making sure that all possible ideas that I can think of are laid out.

**Comment by max-sixty** [comment]
yes, I think that could fit into a #456; I think it's too large a change to do just because of the linguist conflict though

is there any way to hint to linguist which file type it is? it's fairly easy to discriminate between the snapshots...

**Comment by spenserblack** [comment]
> is there any way to hint to linguist which file type it is?

Without a PR into Linguist to add an "Insta Snapshot" language, I think the best thing available would just be to inform Linguist that your files are *not* Jest snapshots with a language override in `.gitattributes`. I don't think any languages besides plain text are really applicable, since pretty much anything can go after the YAML front matter in an Insta snapshot AFAIK.

Since you mentioned #456, I'll share this as a possible idea: have different "sub extensions" for the type of snapshot -- for example, if a JSON snapshot was called `my_test.json.insta.snap`, that filename provides a lot of information for tooling:
- This snapshot is generated by Insta, not Jest
- JSON syntax highlighting can be used after the YAML front matter in the snapshot

---

BTW, even if you use a language override for Linguist, GitHub's search, which uses go-enry for indexing IIRC, will still identify these `*.snap` files as Jest snapshots.

**Comment by max-sixty** [comment]
thanks!

> Without a PR into Linguist to add an "Insta Snapshot" language

is this viable? if anyone want to take a pass, happy to support from here...

> I think the best thing available would just be to inform Linguist that your files are _not_ Jest snapshots with a language override in `.gitattributes`.

yes, ofc fine if users want to do that; though it would need to come from each user rather than `insta`....

ty

**Comment by spenserblack** [comment]
> is this viable?

Based on [this search](https://github.com/search?q=path%3A%2F%5ECargo%5C.toml%24%2F%20%2Finsta%20%3D%2F&type=code), I believe the popularity requirements are met.

Making a PR into Linguist will require the following things:

- A new entry in `languages.yaml` with the language's name, type, and some other attributes
- A submodule pointing towards a repo containing a textmate-compatible grammar
- A sample file
- A heuristic to distinguish Jest snapshots and Insta snapshots

Making PRs into Linguist is generally pretty easy IMO

**Comment by max-sixty** [comment]
ok, I'm happy to give insta's blessing, to the extent that's helpful. (I won't get to it myself though, so folks shouldn't wait for me). ty...

**Comment by Enselic** [comment]
Teaching GitHub linguist about `.snap` files will not be sufficient because GitHub does not respect that when diffing.

See https://github.com/trishume/syntect/pull/592/files where the `.snap` should be diffed like a `Text` file, but it is still syntax highlighted (which is a bug).

But if you look at the file outside of diff context it is not syntax highlighted (which is expected): https://github.com/trishume/syntect/blob/24d48292b0805c6353e94b7561af8c1d54815886/tests/snapshots/public_api__public_api.snap.
